src/debug.c

```diff
@@ -129,6 +129,8 @@ void p11prov_debug_mechanism(P11PROV_CTX *ctx, CK_SLOT_ID slotid,
             }
         }
     }
+    /* XXX Not fatal */
+    ERR_clear_error();
```

Comment:
I am confused by this clear here.

Generally we should set a new error stack up when we enter a function exposed to OpenSSL, and then pop/clear it if we need to suppress any error.

This clearing seem to happen in the middle of a call to a function that is not aware of what error stack is available.

If we want to always prevent any errors from showing up from this function I think we should set up a stack pointer at the start and just pop it at the end, otherwise we risk clearing previous errors that we did want to report.

See the functions: p11prov_set_error_mark(), p11prov_pop_error_to_mark(), p11prov_pop_error_to_mark() and p11prov_clear_last_error_mark()
An example of their use is in store_fetch() and should be extended to other places as needed.

Reply:
Thank you for the pointer. I did not  see these. Indeed, these are the functions I was looking for!